lib/Web/Machine/FSM/States.pm

```diff
@@ -65,10 +65,15 @@ sub _ensure_quoted_header {
 
 sub _get_acceptable_content_type_handler {
     my ($resource, $request) = @_;
-    my $acceptable = match_acceptable_media_type(
-        ($request->header('Content-Type') || 'application/octet-stream'),
-        $resource->content_types_accepted
-    );
+    my $metadata = _metadata($request);
+    my $acceptable = $metadata->{'acceptable_request_handler'};
+    if ( not defined $acceptable ) {
```

Comment:
Use `!` instead of `not` for consistency. Also, if you're going to store the sub (as I suggested) you can just check `if (!$acceptable)` since a subref is always defined (unless someone does something utterly insane with blessed subrefs and overloading, but that's their problem, not ours).